modules/home-manager.nix

```diff
@@ -0,0 +1,59 @@
+# Licensed under GNU Lesser General Public License v3 or later, see COPYING.
+# Copyright (c) 2019 Alexander Sosedkin and other contributors, see AUTHORS.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.home-manager;
+
+  hmModule = types.submodule ({ name, ... }: {
+    imports = import <home-manager/modules/modules.nix> { inherit lib pkgs; };
+
+    config = {
+      submoduleSupport.enable = true;
+      submoduleSupport.externalPackageInstall = cfg.useUserPackages;
+
+      home.username = config.user.userName;
+      home.homeDirectory = config.user.home;
+    };
+  });
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    home-manager = {
+      config = mkOption {
+        type = types.nullOr hmModule;
+        default = null;
+        description = "Home Manager configuration.";
+      };
+
+      useUserPackages = mkEnableOption ''
```

Comment:
There are two user facing differences:
* In `nix-env -q` you can see that there is `nix-on-droid-path` and `home-manager-path` installed. These packages are simple env's with all packages to install. With enabling this option there would only be one (our) package to be installed.
* If you install the same package via `nix-on-droid` and `home-manager` or packages with files with same name, this could lead to name clashes or maybe `nix-env -i` could throw errors trying to install both env's.

It's a nice feature and I see no disadvantages of it. Maybe we should enable it by default?

Reply:
For consistent behaviour I would prefer disabling this option like in nixos and nix-darwin modules.